airbyte-cdk/python/bin/generate-component-manifest-files.sh

```diff
@@ -21,6 +21,10 @@ function main() {
       --use-title-as-name \
       --disable-timestamp \
       --enum-field-as-literal one
+
+    temp_file=$(mktemp)
+    sed 's/_options:/options:/g' "$ROOT_DIR/$OUTPUT_DIR/$filename_wo_ext.py" > "${temp_file}"
+    mv "${temp_file}" "$ROOT_DIR/$OUTPUT_DIR/$filename_wo_ext.py"
```

Comment:
There is an interesting bug in Pydantic that prevents private fields values from being accessible. This means that we cannot invoke `._options` from the component constructors. This has been widely discussed for what looks like 2+ years on the pydantic repo [issue](https://github.com/pydantic/pydantic/issues/288).

Basically why this happens is that we use the `$options` key in our YAML manifests and the pydantic model generator automatically turns special chars into `_` even the first character, so now Pydantic thinks this is a private field. 

Catherine and I have explored a bunch of different options and stack ranked them:

1. the model generator has an option to emit fields w/o prefixing _
2. post-processing of the schema models in declarative_component_schema.py [what we have here]
3. Find a way to parse `model._options` as the underlying value
4. Use parse_obj() at each level of the models
5. Replace `$options` with `options` in the schema + migrate existing manifests
6. Replace `$options` with options in the schema + under the hood transform `$options` to `option` for incoming manifests

Our ideal solution is that the model generator gets rid of the leading underscore, however I've been unable to find a flag to do that. And I haven't been able to find a way of actually getting the value of `model._options` correctly from the constructors. That is why we're left with option 2.

Reply:
yup will add!